src/Umbraco.Web/Models/Mapping/DefaultTemplateResolver.cs

```diff
@@ -8,17 +10,24 @@ internal class DefaultTemplateResolver : IValueResolver<IContent, ContentItemDis
     {
         public string Resolve(IContent source, ContentItemDisplay destination, string destMember, ResolutionContext context)
         {
-            if (source == null || source.Template == null) return null;
+            if (source == null)
+                return null;
 
-            var alias = source.Template.Alias;
+            // If no template id was set...
+            if (!source.TemplateId.HasValue)
+            {
+                // ... and no default template is set, return null...
+                if (string.IsNullOrWhiteSpace(source.ContentType.DefaultTemplate?.Alias))
+                    return null;
 
-            //set default template if template isn't set
-            if (string.IsNullOrEmpty(alias))
-                alias = source.ContentType.DefaultTemplate == null
-                    ? string.Empty
-                    : source.ContentType.DefaultTemplate.Alias;
+                // ... otherwise return the content type default template alias.
+                return source.ContentType.DefaultTemplate?.Alias;
+            }
+            
+            var fileService = DependencyResolver.Current.GetService<IFileService>();
```

Comment:
I and some others think this is a anti pattern (http://blog.ploeh.dk/2010/02/03/ServiceLocatorisanAnti-Pattern/)
You should be able to inject IFileService in the constructor here and in `ContentMapperProfile`. Then you just use LightInject to handle it.

Reply:
@bergmania I tried ctor and parameter injection and they didn't work, unfortunately!